src/certificate.rs

```diff
@@ -178,6 +296,107 @@ impl<'a> TryFrom<&'a [u8]> for Certificate {
 }
 
 impl Certificate {
+    /// Creates a new self-signed certificate for the given key. Writes the resulting
+    /// certificate to the slot before returning it.
+    pub fn generate_self_signed(
+        yubikey: &mut YubiKey,
+        key: SlotId,
+        serial: BigUint,
```

Comment:
Is there any way for the caller to be able to import this type without having to include the `num_bigint` crate themselves?

Would it make sense to have a newtype wrapper (e.g. `struct Serial<BigUint>`) and make this `serial: impl Into<Serial>`, providing some convenient impls for instantiating serials? (e.g. from a byte array/slice?)

Reply:
SGTM. The reason I used `BigUint` here is simply because that is what `x509_parser` represents a serial number as. In `x509` I only require a slice that is at most 20 bytes (and panic if it is longer).